packages/cloud_firestore/test/cloud_firestore_test.dart

```diff
@@ -455,5 +456,87 @@ void main() {
         expect(document.data, equals(kMockDocumentSnapshotData));
       });
     });
+
+    group('FirestoreMessageCodec', () {
+      const MessageCodec<dynamic> codec = const FirestoreMessageCodec();
+      // Java is only precise to millisecond, Dart to microsecond.
+      // The mismatch between millisecond and microsecond precision will fail this test.
+      final DateTime testTime = new DateTime.fromMillisecondsSinceEpoch(
+          new DateTime.now().millisecondsSinceEpoch);
```

Comment:
You might be able to simplify this using a suitable hard-coded value rather than `DateTime.now()`.